packages/tinacms/src/internalClient/index.ts

```diff
@@ -565,10 +569,14 @@ mutation addPendingDocumentMutation(
     init?: RequestInit
   ): Promise<Response> {
     const headers = init?.headers || {}
+    const token = await this.getToken()
+    if (!token?.id_token) {
```

Comment:
@kldavis4  I am unsure about throwing an error here. There might be cases where you want to not pass the token. For example;
- Using cloudinary locally
- Self hosted locally or want to pass a token some other way (maybe a query param)

If we checked that we where hitting tina.io first then I could definitly see throwing an error.

Sudo code

```ts
if (!token?.id_token && url.contains('tina.io') {
     throw new Error('Authentication required')
}
```

Reply:
If that's the case, that is not a good name for that function

Reply:
Agreed it could definitely be renamed in the future.